lib/ecto_cursor_based_stream.ex

```diff
@@ -73,16 +75,17 @@ defmodule EctoCursorBasedStream do
 
   @doc false
   def call(repo, queryable, options \\ []) do
-    %{max_rows: max_rows, after_cursor: after_cursor, cursor_field: cursor_field} =
-      parse_options(options)
+    %{max_rows: max_rows, after_cursor: after_cursor, cursor_field: cursor_field} = parse_options(options)
 
     Stream.unfold(after_cursor, fn cursor ->
       case get_rows(repo, queryable, cursor_field, cursor, max_rows) do
         [] ->
           nil
 
         rows ->
-          next_cursor = get_last_row_cursor(rows, cursor_field)
+          [last | _] = :lists.reverse(rows)
```

Comment:
how is that better?

Reply:
it is marginally better, in exchange for some memory bits consumed. You can run some tests yourself, but here just in case:

```txt
Name                       ips        average  deviation         median         99th %
:lists.reverse/1      770.27 K        1.30 μs  ±1734.85%        1.08 μs        4.04 μs
List.last/1           604.74 K        1.65 μs   ±214.70%        1.58 μs        2.13 μs

Comparison:
:lists.reverse/1      770.27 K
List.last/1           604.74 K - 1.27x slower +0.36 μs
```

the test above was done for a list with 1000 values in it. Obviously not an exhaustive test, but you get the idea.